src/plone/app/multilingual/browser/add.py

```diff
@@ -32,9 +32,10 @@ def traverse(self, name, ignored):
         ti = ttool.getTypeInfo(name)
         if not IDexterityFTI.providedBy(ti):
             # we are not on DX content
-            baseUrl = self.context.absolute_url()
-            url = '%s/@@add_at_translation?type=%s' % (baseUrl, name)
-            return self.request.response.redirect(url)
+            self.context.REQUEST.set('type', name)
+            view = queryMultiAdapter((self.context, self.context.REQUEST),
+                                         name="add_at_translation")
```

Comment:
When following the `plone.api` style guidelines - which all of us should - you should update:
- Fix indention level
- Prefer single quotes